native_client/deepspeech.cc

```diff
@@ -93,33 +479,47 @@ Model::Model(const char* aModelPath, int aNCep, int aNContext,
     status = ReadBinaryProto(Env::Default(), aModelPath, &mPriv->graph_def);
   }
   if (!status.ok()) {
-    mPriv->session->Close();
-    mPriv->session = NULL;
-    std::cerr << status.ToString() << std::endl;
+    Status closeStatus = mPriv->session->Close();
```

Comment:
All this kind of close code is error prone.

I would be much easier to introduce a little [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) wrapper class that does this in it's destructor.

Reply:
Will do.